DIPs/1NNN-ADR.md

```diff
@@ -28,25 +84,71 @@ embedding the arguments in the string itself.
 
 ## Rationale
 
-While the conventional format string followed by the argument list is fine for
-short strings and a small number of arguments, it tends to break down with longer strings
-that have many arguments. Omitted arguments, extra arguments, and mismatches
-between format specifiers and their corresponding arguments are common errors. Embedding arguments
-in the format strings can eliminate these errors. Readability is improved and the code is visually
-easier to review for correctness.
+This proposal was born from the rejection of DIP1027 that had the desire to have interpolated strings 
+found in other languages and to have a generic capability to use the interpolated syntax with other domains.  For example,
+it should be possible to have an SQL aware transform.
+
+```D
+    --- Put compelling SQL example here... ---
+    auto query = "SELECT field1, field2 FROM table WHERE field1 = :field1Value;";
+    auto stmt = db.prepare( query );
+    stmt.bind( ":field1Value", value1 );
+```
+
+Another example is the generation of HTML tags
+
+```D
+    --- Put compelling HTML example here ---
+    auto string div = "<div id=\"" ~ id ~ "\">"
+```
+
+These examples both benefit from the proposed interpolation syntax because the compiler can
+semantically verify the correctness of the expression while preparing a tuple suitable for arguments
+for subsequent function calls.
+
+```D
+    auto query = i"SELECT field1, field2 FROM table WHERE field1 = ${field1Value};";
```

Comment:
you know what I would secretly love? if it was javascript style and didn't actually use the i prefix but instead did a function name prefix so like `sql"....."` passes the arg to the sql function.

but yeah definitely gotta make it clear in here that you do need to pass it to af unction somehow.

I think if I did a few little edits I can work in most your stuff, my only hesitation is I think Mike Parker already put it as-is into the queue; might be too late. I should have noticed this a couple days ago....